File: src/main/java/LDS/Person/service/WikiService.java
package LDS.Person.service;

import LDS.Person.dto.request.WikiCreateRequest;
import LDS.Person.dto.request.WikiPageQueryRequest;
import LDS.Person.dto.response.LatestWikiSummaryResponse;
import LDS.Person.dto.response.WikiResponse;
import LDS.Person.dto.response.PageResponse;

/**
 * Wiki 业务逻辑接口
 */
public interface WikiService {

    /**
     * 新增 Wiki 记录
     * 
     * @param request 创建请求
     * @return Wiki 响应
     */
    WikiResponse createWiki(WikiCreateRequest request);

    /**
     * 删除 Wiki 记录
     * 
     * @param wikiId Wiki ID
     * @return true 删除成功，false 删除失败（记录不存在）
     */
    boolean deleteWiki(Long wikiId);

    /**
     * 分页查询 Wiki 列表
     * 支持多条件过滤和排序
     * 
     * @param request 分页查询请求
     * @return 分页响应
     */
    PageResponse<WikiResponse> pageQuery(WikiPageQueryRequest request);

    /**
     * 检查 Wiki 键名是否已存在
     * 
     * @param keyName Wiki 键名
     * @return true 表示键名已存在，false 表示不存在
     */
    boolean isKeyNameExists(String keyName);

    /**
     * 根据 Wiki ID 查询完整内容
     * 
     * @param wikiId Wiki ID
     * @return Wiki 响应，如果不存在则返回 null
     */
    WikiResponse getWikiById(Long wikiId);

    /**
     * 更新 Wiki 记录
     * 
     * @param wikiId  Wiki ID
     * @param request 更新请求
     * @return 更新后的 Wiki 响应
     */
    WikiResponse updateWiki(Long wikiId, LDS.Person.dto.request.WikiUpdateRequest request);

    /**
     * 随机获取一个 Wiki 的完整内容
     * 
     * @return 随机选中的 Wiki 响应，如果不存在则返回 null
     */
    WikiResponse getRandomWiki();

    /**
     * 随机获取四个 Wiki 的列表（分页格式）
     * 
     * @return 包含四个随机 Wiki 的分页响应
     */
    PageResponse<WikiResponse> getRandomWikis();

    /**
     * 获取最新更新过的 Wiki 元信息
     * 
     * @return 最新更新 Wiki 的摘要数据
     */
    LatestWikiSummaryResponse getLatestUpdatedWiki();
}

File: src/Services/WikiService.cs
using SidusX.DTOs.Request;
using SidusX.DTOs.Response;
using SidusX.Models;
using SidusX.Repositories;

namespace SidusX.Services;

public class WikiService
{
    private readonly WikiRepository _wikiRepo;
    private readonly WikiHistoryRepository _historyRepo;
    private readonly UserRepository _userRepo;
    private readonly ILogger<WikiService> _logger;

    public WikiService(WikiRepository wikiRepo, WikiHistoryRepository historyRepo,
        UserRepository userRepo, ILogger<WikiService> logger)
    {
        _wikiRepo = wikiRepo;
        _historyRepo = historyRepo;
        _userRepo = userRepo;
        _logger = logger;
    }

    public async Task<WikiResponse> CreateWikiAsync(WikiCreateRequest request)
    {
        if (string.IsNullOrWhiteSpace(request.KeyName)) throw new ArgumentException("key_name 不能为空");
        if (string.IsNullOrWhiteSpace(request.Texts)) throw new ArgumentException("texts 不能为空");
        if (await _wikiRepo.IsKeyNameExistsAsync(request.KeyName)) throw new ArgumentException($"key_name '{request.KeyName}' 已存在");

        var now = DateTime.Now;
        var wiki = new Wiki
        {
            KeyName = request.KeyName, Texts = request.Texts, Tags = request.Tags,
            Version = 1.0, CreateTime = now, CreateUser = request.CreateUser ?? "system",
            UpdateTime = now, UpdateUser = request.UpdateUser ?? request.CreateUser ?? "system"
        };
        wiki.WikiId = await _wikiRepo.InsertAsync(wiki);
        return ToResponse(wiki);
    }

    public async Task<bool> DeleteWikiAsync(long wikiId)
    {
        if (wikiId <= 0) throw new ArgumentException("wikiId 不合法");
        return await _wikiRepo.DeleteAsync(wikiId);
    }

    public async Task<WikiResponse?> GetByIdAsync(long wikiId)
    {
        var wiki = await _wikiRepo.GetByIdAsync(wikiId);
        return wiki == null ? null : ToResponse(wiki);
    }

    public async Task<WikiResponse?> GetByKeyNameAsync(string keyName)
    {
        var wiki = await _wikiRepo.GetByKeyNameAsync(keyName);
        return wiki == null ? null : ToResponse(wiki);
    }

    public async Task<bool> IsKeyNameExistsAsync(string keyName) =>
        await _wikiRepo.IsKeyNameExistsAsync(keyName);

    public async Task<WikiResponse?> UpdateWikiAsync(WikiUpdateRequest request)
    {
        if (request.WikiId <= 0) throw new ArgumentException("wikiId 不合法");
        var existing = await _wikiRepo.GetByIdAsync(request.WikiId);
        if (existing == null) return null;

        // Save to history first
        await _historyRepo.InsertAsync(new WikiHistory
        {
            WikiId = existing.WikiId, KeyName = existing.KeyName, Texts = existing.Texts,
            Tags = existing.Tags, Version = existing.Version, CreateTime = existing.CreateTime,
            CreateUser = existing.CreateUser, UpdateTime = existing.UpdateTime,
            UpdateUser = existing.UpdateUser, BackupTime = DateTime.Now
        });

        await _wikiRepo.UpdateAsync(existing.WikiId,
            request.Texts ?? existing.Texts,
            request.Tags ?? existing.Tags,
            request.UpdateUser);

        var updated = await _wikiRepo.GetByIdAsync(existing.WikiId);
        return updated == null ? null : ToResponse(updated);
    }

    public async Task<PageResponse<WikiResponse>> PageQueryAsync(WikiPageQueryRequest request)
    {
        var (items, total) = await _wikiRepo.PageQueryAsync(request);
        return new PageResponse<WikiResponse>
        {
            Total = total, Page = request.Page, PageSize = request.PageSize,
            Data = items.Select(ToResponse).ToList()
        };
    }

    public async Task<List<LatestWikiSummaryResponse>> GetLatestAsync(int limit = 10)
    {
        var items = await _wikiRepo.GetLatestAsync(limit);
        return items.Select(w => new LatestWikiSummaryResponse
        {
            WikiId = w.WikiId, KeyName = w.KeyName, UpdateTime = w.UpdateTime, UpdateUser = w.UpdateUser
        }).ToList();
    }

    private static WikiResponse ToResponse(Wiki w) => new()
    {
        WikiId = w.WikiId, KeyName = w.KeyName, Texts = w.Texts, Tags = w.Tags,
        Version = w.Version, CreateTime = w.CreateTime, CreateUser = w.CreateUser,
        UpdateTime = w.UpdateTime, UpdateUser = w.UpdateUser
    };
}
